tracing-core/src/dispatch.rs

```diff
@@ -510,6 +510,7 @@ pub(crate) fn get_global() -> &'static Dispatch {
     unsafe {
         // This is safe given the invariant that setting the global dispatcher
         // also sets `GLOBAL_INIT` to `INITIALIZED`.
+        #[allow(static_mut_refs)]
```

Comment:
We'd prefer to have these changes to adhere to new clippy lints split into a separate PR to go in first. This keeps the clippy changes isolated from the actual fix.

On the up side, you get 2 PRs on `tracing` for the price of 1!

Reply:
#2948